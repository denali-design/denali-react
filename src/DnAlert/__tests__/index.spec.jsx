/**
 * Copyright 2022, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { DnAlert, DnAlertStatus } from '../../index'; // src/index.tsx

describe('DnAlert Tests', () => {
    it('should export DnAlertStatus', () => {
        expect.assertions(1);

        expect(DnAlertStatus).toStrictEqual({
            default: 'default',
            info: 'info',
            warning: 'warning',
            success: 'success',
            danger: 'danger',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnAlert icon="domain" title="This is a title" />);

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnAlert
                icon="stop-urgent"
                title="This is the title"
                context="The message provides users with context of the alert."
                status={DnAlertStatus.danger}
                isBlock
                className="test--class"
                tabIndex={0}
                onClose={() => ({})}
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should fire onClose callbacks', () => {
        expect.assertions(2);

        const onClose = jest.fn();
        const onCloseKeyDown = jest.fn();
        const wrapper = mount(
            <DnAlert
                icon="stop-urgent"
                title="This is the title"
                context="The message provides users with context of the alert."
                status={DnAlertStatus.danger}
                isBlock
                className="test--class"
                onClose={onClose}
                onCloseKeyDown={onCloseKeyDown}
            />,
        );

        const closeElm = wrapper.find('.close');

        closeElm.simulate('click');

        expect(onClose).toHaveBeenCalledTimes(1);

        closeElm.simulate('keydown', { key: 'Enter' });

        expect(onCloseKeyDown).toHaveBeenCalledTimes(1);
    });
});
