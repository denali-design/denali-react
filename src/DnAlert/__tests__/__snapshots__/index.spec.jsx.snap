// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnAlert Tests should render correctly with custom props 1`] = `
<div
  class="alert is-danger is-block test--class"
>
  <i
    class="d-icon d-stop-urgent is-default"
  />
  <div>
    <h5>
      This is the title
    </h5>
    <p>
      The message provides users with context of the alert.
    </p>
  </div>
  <span
    class="link close is-secondary"
    role="button"
    tabindex="0"
  >
    <i
      class="d-icon d-close is-default is-sub"
    />
  </span>
</div>
`;

exports[`DnAlert Tests should render with default props 1`] = `
<div
  class="alert is-default"
>
  <i
    class="d-icon d-domain is-default"
  />
  <div>
    <h5>
      This is a title
    </h5>
  </div>
</div>
`;
